src/python/test/test_simplex_tree.py

```diff
@@ -399,3 +400,17 @@ def test_boundaries_iterator():
 
     with pytest.raises(RuntimeError):
         list(st.get_boundaries([6])) # (6) does not exist
+
+# insert with NaN must not modify filtration values
+# make_filtration_non_decreasing considers NaN small
+def test_nan():
+    st = SimplexTree()
+    st.insert([1], 2.)
+    st.insert([4], 5.)
+    st.insert([2], -1.)
```

Comment:
```suggestion
    st.insert([2], -1.)
    st.insert([1,2], 2.)
```
Perhaps already add a single edge (with an admissible value) to make sure that both inserting `nan` and `make_filtration_non_decreasing()` do not play any role there

Reply:
I inserted an extra edge.